### ISSUE TITLE
Implement mod dependencies

### ISSUE THREAD

**Issue by Puros** [flareteam/flare-engine#733]
Allow mods to establish dependency on other mods and to (initially) override them in priority in the config menu.

Excerpt from http://opengameart.org/forumtopic/health-potion-gold where this thread started:
# alpha_demo/settings.txt

depends=fantasycore
description=The demo game for Flare

PRIORITY:
Activating both fantasycore and alpha_game would result in alpha_game taking the priority position regardless of whether it was selected first or last.

INCLUSION:
Additionally, when alpha_game is selected for activation, fantasycore should be selected as well (if it is not already) since it is a dependency.

WARNING:
Perhaps a warning icon or red text for "missing" dependencies when they are disabled manually or order is changed in the "active mods" list.

TL;DR
Give the gamer a working game by default.
